meta-facebook/minerva-ag/src/shell/plat_soc_pwron_reset_shell.c

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <shell/shell.h>
+#include <stdlib.h>
+#include <logging/log.h>
+#include "plat_isr.h"
+#include "plat_i2c.h"
+
+#define AEGIS_CPLD_ADDR (0x4C >> 1)
+
+LOG_MODULE_REGISTER(plat_soc_pwron_reset_shell, LOG_LEVEL_DBG);
+
+static int cmd_soc_pwron_reset_set(const struct shell *shell, size_t argc, char **argv)
+{
+	uint8_t data = 0;
+	if (!strcmp(argv[0], "override")) {
+		if (!strcmp(argv[1], "1")) {
+			data = 0xFF;
+			if (!plat_i2c_write(I2C_BUS5, AEGIS_CPLD_ADDR, 0x00, &data, 1)) {
+				shell_error(shell, "plat soc_pwron_reset set failed");
+				return -1;
+			}
+		} else if (!strcmp(argv[1], "0")) {
+			data = 0xEF;
+			if (!plat_i2c_write(I2C_BUS5, AEGIS_CPLD_ADDR, 0x00, &data, 1)) {
+				shell_error(shell, "plat soc_pwron_reset set failed");
+				return -1;
+			}
+		} else {
+			shell_warn(
+				shell,
+				"Help: soc_pwron_reset override <drive-level> should only accept 0 or 1");
+			return -1;
+		}
+	} else if (!strcmp(argv[0], "passthru")) {
+		data = 0xFF;
+		if (!plat_i2c_write(I2C_BUS5, AEGIS_CPLD_ADDR, 0x00, &data, 1)) {
+			shell_error(shell, "plat soc_pwron_reset set failed");
+			return -1;
+		}
+	} else {
+		shell_warn(shell, "Help: soc_pwron_reset <command> <drive-level>");
```

Comment:
Update this so that it's clear with the valid commands are. It looks like drive level is optional (for the pass through case) so let's make that clear